File: sample/goclient/proto/dependtest.go
// sample/proto/Depend.proto

package proto

import (
	protopack "github.com/yazawa-ichio/protoc-gen-msgpack/lib/golang"
)

type DependTest struct {
	Dep *Forum_PostData
}

// Pack Serialize Message
func (m *DependTest) Pack() ([]byte, error) {
	return protopack.Pack(m)
}

// Write Serialize Message
func (m *DependTest) Write(w protopack.Writer) error {
	// Write Map Length
	err := w.WriteMapHeader(1)
	if err != nil {
		return err
	}
	
	// Write dep
	err = w.WriteTag(1)
	if err != nil {
		return err
	}
	err = w.WriteMessage(m.Dep)
	if err != nil {
		return err
	}
	return err
}

// Unpack Serialize Message
func (m *DependTest) Unpack(buf []byte) (error) {
	return protopack.Unpack(m, buf)
}

// Read Deserialize Message
func (m *DependTest) Read(r protopack.Reader) error {
	// Read Map Length
	len, err := r.ReadMapHeader()
	if err != nil {
		return err
	}

	for i := uint(0); i < len; i++ {

		// Read Tag
		tag, err := r.ReadTag()
		if err != nil {
			return err
		}

		switch tag {
		case 1: // Read dep
			isNil, err := r.NextFormatIsNull()
			if err != nil {
				return err
			}
			if isNil {
				m.Dep = nil
				err = r.ReadNil()
			} else {
				m.Dep = &Forum_PostData{}
				err = m.Dep.Read(r)
			}
			if err != nil {
				return err
			}
			break
		default:
			err = r.Skip()
			if err != nil {
				return err
			}
			break
		}
	}
	return err
}

File: sample/client/proto/DependTest.cs
// sample/proto/Depend.proto
using ILib.ProtoPack;
using System.Collections.Generic;
[System.Serializable]
public partial class DependTest : IMessage
{
	public Forum.PostData Dep;

	#region Serialization

	/// <summary>
	/// Serialize Message
	/// </summary>
	public void Write(IWriter w)
	{
		// Write Map Length
		w.WriteMapHeader(1);
		
		// Write dep
		w.WriteTag(1);
		w.Write(Dep);
	}

	/// <summary>
	/// Deserialize Message
	/// </summary>
	public void Read(IReader r)
	{
		// Read Map Length
		var len = r.ReadMapHeader();

		for (var i = 0; i < len; i++)
		{
			var tag = r.ReadTag();
			switch(tag) {
			case 1: // Read dep
				Dep = r.ReadMessage<Forum.PostData>();
				break;
			default:
				r.Skip();
				break;
			}
		}
	}
	#endregion

}
